content/essentials/class-style-bindings.mdx

```diff
@@ -0,0 +1,90 @@
+import { Aside } from "~/components/configurable/Aside";
+
+# Class & Style Bindings
+
+Similar to HTML, Solid uses `class` and `style` attributes to style elements via [CSS (Cascading Style Sheets)](https://developer.mozilla.org/en-US/docs/Glossary/CSS).
+
+* **Class Attribute**: Enables styling one or more elements through CSS rules.
+* **Style Attribute**: Inline styles that style single elements.
+
+## Inline styling
+
+The `style` attribute lets you style a single element and define CSS variables dynamically during runtime. 
+Solid allows `style` in both string (e.g. `<div style="color: red;"` /> and object forms  (e.g. `<div style={{ "color": "red" }}>` ).
+
+If you are styling using the object form, the keys will require dash-case form and any units must be explicitly provided:
+
+```jsx
+<div style={{
+  color: `red`,
+ "font-weight": 800,
+ "font-size": '2rem'
+}}
+>
+  Some Text
+</div>
+```
+
+Inline styles come in handy for rapid prototyping, setting dynamic CSS variables, and overriding classes. 
+However, because Solid uses JSX, it is not recommended to use the `style` tag within the JSX code directly. 
+Doing this clutters the component logic, making the code less readable and harder to maintain over time.
+
+## Classes
+
+Through the `class` attribute, you can style elements using CSS. The `class` attribute offers self-documentation and performance advantages over `style`.
+
+For a structured approach, pair CSS files with their respective component files. 
+Importing these files will globally scopes them.
+This means that the elements using those classes will inherit the styles automatically across the whole project, not just the ones located in the file they are imported into.
+
+```jsx
+import "./Card.css";
+function Card() {
+  // ...
+}
+```
+
+In scenarios where you might want to conditionally set classes, Solid offers the built-in [`classList`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) JSX attribute. 
+This attribute takes an object where the key represents the class name(s) and the value is a boolean expression. 
+When the value is to true, the class is applied; when false, it is removed.
+
+```tsx
+  const [current, setCurrent] = createSignal("foo");
+
+    <button
+     class={current() === 'foo' ? 'selected' : ''}
+     onClick={() => setCurrent('foo')}
+    >foo</button>
+```
```

Comment:
Maybe this example should move up above `classList`, and then there should be an example of `classList` here, like `classList: {selected: current() === 'foo'}`?

I note in the [old API docs](https://www.solidjs.com/docs/latest/api#classlist) that `classList` doesn't work with spreads or in `Dynamic`, so perhaps that's worth mentioning here too. Also there are extensive warnings about mixing `class` and `classList` which are probably worth repeating.

I'm also confused about the difference between this subsection and the next one on Dynamic styling. It seems like this example is already dynamic? Maybe we want some transition like "One great thing classes enable is dynamic styling..."?

Reply:
Looks great!! Thank you.